Code/Framework/AzCore/AzCore/DOM/DomPrefixTree.inl

```diff
@@ -62,70 +62,137 @@ namespace AZ::Dom
     }
 
     template<class T>
-    void DomPrefixTree<T>::VisitPath(const Path& path, PrefixTreeMatch match, const VisitorFunction& visitor) const
+    void DomPrefixTree<T>::VisitPath(const Path& path, const VisitorFunction& visitor, PrefixTreeTraversalFlags flags)
     {
-        const Node* rootNode = GetNodeForPath(path);
+        const bool processOurPath = (flags & PrefixTreeTraversalFlags::ExcludeExactPath) == PrefixTreeTraversalFlags::None;
+        const bool processOurParents = (flags & PrefixTreeTraversalFlags::ExcludeParentPaths) == PrefixTreeTraversalFlags::None;
+        const bool processOurChildren = (flags & PrefixTreeTraversalFlags::ExcludeChildPaths) == PrefixTreeTraversalFlags::None;
+
+        Node* rootNode = processOurParents ? &m_rootNode : GetNodeForPath(path);
         if (rootNode == nullptr)
         {
             return;
         }
 
-        if ((match == PrefixTreeMatch::ExactPath || match == PrefixTreeMatch::PathAndSubpaths) && rootNode->m_data.has_value())
+        Path currentPath = processOurParents ? Path() : path;
+        struct PopPathEntry
         {
-            visitor(path, rootNode->m_data.value());
-        }
+        };
+        using StackEntry = AZStd::variant<Node*, PathEntry, PopPathEntry>;
 
-        if (match == PrefixTreeMatch::ExactPath)
-        {
-            return;
-        }
+        const bool depthFirst = (flags & PrefixTreeTraversalFlags::TraverseMostToLeastSpecific) != PrefixTreeTraversalFlags::None;
+        AZStd::stack<AZStd::pair<AZ::Dom::Path, Node*>> depthStack;
 
-        Path currentPath = path;
-        struct PopPathEntry
+        auto maybeVisitEntry = [&](Node* node) -> bool
         {
+            if (!node->m_data.has_value())
+            {
+                return true;
+            }
+            if ((processOurParents && currentPath.Size() < path.Size()) || (processOurPath && currentPath.Size() == path.Size()) ||
+                (processOurChildren && currentPath.Size() > path.Size()))
```

Comment:
Does the DomPath have an IsPrefix function on it that could be used instead of .Size()?
Ala how AZ::IO::Path has [IsRelativeTo](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/Path/Path.h#L179-L180)

Reply:
It does not, I agree that something like a `StartsWith` method would be good there, but in this case just comparing sizes is safe and likely slightly faster.